pyramid_swagger/api.py

```diff
@@ -101,26 +101,206 @@ def view_for_api_declaration(request):
+    elif len(current_path_url.scheme) > 0:
+        target_scheme = current_path_url.scheme
+    else:
+        target_scheme = 'file'
+
+    if target_scheme == 'file':  # targeting a local file
+        # if not absolute path, make it absolute
+        if not target_url.path.startswith('/'):
+            # if path is empty, then is targeting to the current file
+            filename = target_url.path if len(target_url.path) > 0 \
+                else os.path.basename(current_path_url.path)
+            target_url = urlparse('{path}#{fragment}'.format(
+                path=os.path.join(
+                    spec_dir,
+                    os.path.dirname(current_path_url.path),
+                    filename,
+                ),
+                fragment=target_url.fragment,
+            ))
+        return target_url
+
+    elif target_scheme in ['http', 'https']:
+        remote_path = os.path.abspath(os.path.join(
+            current_path_url.path if len(current_path_url.path) > 0 else '/',
+            target_url.path
+        ))
+        return urlparse('{scheme}://{hostname}{path}#{fragment}'.format(
+            scheme=target_scheme,
+            hostname=target_url.hostname,
+            path=remote_path,
+            fragment=target_url.fragment,
+        ))
+    else:
+        return urlparse(relative_path)
+
+
+def _relpath(path, start):
+    """Return a relative version of a path
+    NOTE: Code duplicated from Python 2.7.9 implementation because the
+    default implementation available on Python 2.6.9 is bugged.
+    """
+
+    start_list = [x for x in os.path.abspath(start).split(os.path.sep) if x]
+    path_list = [x for x in os.path.abspath(path).split(os.path.sep) if x]
+
+    # Work out how much of the filepath is shared by start and path.
+    i = len(os.path.commonprefix([start_list, path_list]))
+
+    rel_list = [os.path.pardir] * (len(start_list) - i) + path_list[i:]
+
+    return os.path.join(*rel_list)
+
+
+def _get_target(spec, target, current_path=''):
+    # Note: we assume that swagger.json file is always a server-local file
+    if target == '':
+        raise ValueError('Empty target')
+
+    target_url = _get_absolute_link(spec, target, current_path)
+    target_scheme = 'file' if target_url.scheme == '' else target_url.scheme
+
+    if target_scheme == 'file':
+        spec_dir = os.path.dirname(urlparse(spec.origin_url).path)
+        target_url = urlparse('{path}#{fragment}'.format(
+            path=_relpath(target_url.path, spec_dir),
+            fragment=target_url.fragment,
+        ))
+
+    if target_scheme in ['file', 'http', 'https']:
+        return target_url
+    else:
+        raise ValueError(
+            'Invalid target: {target}'.format(target=target)
+        )
+
+
+def _marshal_target(target_url):
+    target_scheme = target_url.scheme
+    if len(target_url.path) > 0 and \
+            target_scheme in ['', 'file', 'http', 'https']:
+        value = _low_level_translate(
```

Comment:
If we're just immediately returning value, I'd do it on this line instead of the temporary variable